test/FilesystemManagementTests.h

```diff
@@ -0,0 +1,169 @@
+            }
+            UtilityFileDetect.close();
+        }// File Detection Test using ifstream
+        std::ofstream UtilityFile;
+        UtilityFile.open("./UtilityTestFile.txt",std::ios_base::out | std::ios_base::trunc);
+        UtilityFile << "I exist!";
+        UtilityFile.close();
+
+        TEST_EQUAL("FileExists(const_StringView)-PassCheck",
+                   true,Filesystem::FileExists("UtilityTestFile.txt"));
+        TEST_EQUAL("CopyFile(const_StringView,const_StringView)-Fresh",
+                   Filesystem::ModifyResult::Success,
+                   Filesystem::CopyFile("UtilityTestFile.txt","UtilityTestCopy.txt",false));
+        TEST_EQUAL("CopyFile(const_StringView,const_StringView)-Exists",
+                   true,Filesystem::FileExists("UtilityTestCopy.txt"));
+        TEST_EQUAL("CopyFile(const_StringView,const_StringView)-DuplicateFail",
+                   Filesystem::ModifyResult::AlreadyExists,
+                   Filesystem::CopyFile("UtilityTestCopy.txt","UtilityTestFile.txt",true));
+        TEST_EQUAL("MoveFile(const_StringView,const_StringView)-CreateDest",
+                   Filesystem::ModifyResult::Success,
+                   Filesystem::CreateDirectory("MoveTarget/"));
+        TEST_EQUAL("MoveFile(const_StringView,const_StringView)-ActualMove",
+                   Filesystem::ModifyResult::Success,
+                   Filesystem::MoveFile("UtilityTestCopy.txt","MoveTarget/UtilityTestCopy.txt",true));
+        TEST_EQUAL("MoveFile(const_StringView,const_StringView)-SourceDoesntExist",
+                   false,Filesystem::FileExists("UtilityTestCopy.txt"));
+        TEST_EQUAL("MoveFile(const_StringView,const_StringView)-DestExists",
+                   true,Filesystem::FileExists("MoveTarget/UtilityTestCopy.txt"));
+        TEST_EQUAL("MoveFile(const_StringView,const_StringView)-Rename",
+                   Filesystem::ModifyResult::Success,
+                   Filesystem::MoveFile("MoveTarget/UtilityTestCopy.txt","MoveTarget/RenamedCopy.txt",true));
+        TEST_EQUAL("RemoveFile(const_StringView)-DeleteMoved",
+                   Filesystem::ModifyResult::Success,
+                   Filesystem::RemoveFile("MoveTarget/RenamedCopy.txt"));
+        TEST_EQUAL("RemoveFile(const_StringView)-VerifyMoved",
+                   false,Filesystem::FileExists("MoveTarget/RenamedCopy.txt"));
+        TEST_EQUAL("RemoveFile(const_StringView)-DeleteOriginal",
+                   Filesystem::ModifyResult::Success,
+                   Filesystem::RemoveFile("UtilityTestFile.txt"));
+        TEST_EQUAL("RemoveFile(const_StringView)-VerifyDeleted",
+                   false,Filesystem::FileExists("UtilityTestFile.txt"));
+        TEST_EQUAL("MoveFile(const_StringView,const_StringView)-DestroyDest",
+                   Filesystem::ModifyResult::Success,
+                   Filesystem::RemoveDirectory("MoveTarget"));
+    }// Basic File Management
+
+    {// Symlinks
+        // No tests for symlinks exist yet.
```

Comment:
You could make a file, put something in it, then make symlink to the file, and verify the contents match what you wrote. Same idea with directory symlinks.

This would also let us choose and test what the non-symlink functions do when fed symlinks.

Reply:
I have updated quite a bit of functionality relating to symlinks.  Including the addition of tests.